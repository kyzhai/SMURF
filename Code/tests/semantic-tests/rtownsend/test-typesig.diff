14c14
< 	ID: print x :: a -> a
---
> 	ID: print  :: Unknown
