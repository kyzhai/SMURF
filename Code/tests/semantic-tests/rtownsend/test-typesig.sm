/* We are testing the following for success:
	 1. Defining the type for variables
	 2. Defining some function types

	 */

a :: Int
b :: Bool
c :: Note
d :: Beat
e :: Chord
f :: System
g :: [Int]
h :: [Int] -> [[Int]] -> Bool
i :: a -> a -> b

main = []
