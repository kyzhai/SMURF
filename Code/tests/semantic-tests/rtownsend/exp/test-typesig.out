Program: 
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: i  :: a -> a -> b
	ID: h  :: [Int] -> [[Int]] -> Bool
	ID: g  :: [Int]
	ID: f  :: System
	ID: e  :: Chord
	ID: d  :: Beat
	ID: c  :: Note
	ID: b  :: Bool
	ID: a  :: Int
	ID: print x :: a -> a
	ID: random  :: Int
	ID: head h : t :: [a] -> a
	ID: tail h : t :: [a] -> [a]

